pkg/volume/iscsi/iscsi_util.go

```diff
@@ -740,23 +751,29 @@ func (util *ISCSIUtil) detachISCSIDisk(exec mount.Exec, portals []string, iqn, i
 		}
 		klog.Infof("iscsi: log out target %s iqn %s iface %s", portal, iqn, iface)
 		out, err := exec.Run("iscsiadm", logoutArgs...)
+		err = ignoreExitCodes(err, exit_ISCSI_ERR_NO_OBJS_FOUND, exit_ISCSI_ERR_SESS_NOT_FOUND)
 		if err != nil {
 			klog.Errorf("iscsi: failed to detach disk Error: %s", string(out))
+			return err
 		}
 		// Delete the node record
 		klog.Infof("iscsi: delete node record target %s iqn %s", portal, iqn)
 		out, err = exec.Run("iscsiadm", deleteArgs...)
+		err = ignoreExitCodes(err, exit_ISCSI_ERR_NO_OBJS_FOUND, exit_ISCSI_ERR_SESS_NOT_FOUND)
 		if err != nil {
 			klog.Errorf("iscsi: failed to delete node record Error: %s", string(out))
+			return err
```

Comment:
The changes LGTM, specially the device flushing; however, IDK iSCSI well enough to be sure these are the only exit codes we could/should ignore.

For example, looking through the exit codes it seems we should always etry on `ISCSI_ERR_AGAIN` and `ISCSI_ERR_BUSY`, but not for `ISCSI_ERR_ACCESS` and `ISCSI_ERR_INVAL`. To make it difficult, these codes are tied to a specific set of commands.

Reply:
Anyway, I think it's OK to retry regardless

Reply:
I don't see ISCSI_ERR_AGAIN and ISCSI_ERR_BUSY in my man page (iscsiadm version 6.2.0.876-1)

Reply:
I saw `ISCSI_ERR_NO_OBJS_FOUND` for deleting already deleted session and I think I saw `ISCSI_ERR_SESS_NOT_FOUND` in some log.

Reply:
I was looking at the [iscsi_err.h](https://github.com/open-iscsi/open-iscsi/blob/7d121d12ad6ba7783308c25ffd338a9fa0cc402b/include/iscsi_err.h#L7) file, so maybe these exit codes aren't returned by iscsiadm.

Reply:
There are a bunch of different things that can be returned from iscsiadm that may or may not matter.  One way to approach this is ignore any error, and do a verify that things were cleaned up, if it has the error is irrelevant.

Reply:
> verify that things were cleaned up

I can check that iface and node need to be deleted easily (`-m [node|iface] -o show`).

I am not sure if it's possible to check if `iscsiadm -m node -p 127.0.0.1 -T iqn.2003-01.io.k8s:e2e.volume-8751 --logout` is needed. `iscsiadm -m node -o show` is the same before and after logout. How can I map node / iqn to a session without too much parsing of `iscsiadm -m session -o show -P 2`?

Reply:
Yeah, I was thinking of using the approach to check for sessions; I may be overthinking this though.